src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/TaskMethodInvoker.cs

```diff
@@ -80,9 +80,8 @@ public object InvokeEnd(object instance, out object[] outputs, IAsyncResult resu
             {
                 //AsyncMethodInvoker.GetActivityInfo(ref activity, ref boundOperation);
 
-                Task<Tuple<object, object[]>> invokeTask = result as Task<Tuple<object, object[]>>;
-
-                if (invokeTask == null)
+                var asyncResult = result as TaskHelpers.AsyncResult<Tuple<object, object[]>>;
```

Comment:
I did not want to rewrite this method more than necessary but I think It would benefit from calling `ToApmEnd` instead which would involve moving the processing of the AggregateException "ae" into a try/catch block around the call to `ToApmEnd`

Reply:
The key thing is does it still work. I can't remember what range of tests we have around this but before making any drastic changes, we need to make sure there's tests for:
1. Task based method which completes synchronously
2. Task based method which completes async
3. Task<TResult> based method which completes synchronously
4. Task<TResult> based method which completes async
5. Task based method which returns a synchronously completed Task which denies child attach (continuation must execute async)
6. Task<TResult> based method which returns a synchronously completed Task which denies child attach (continuation must execute async)
7. Task based method which completes async and the returned Task denies child attach (continuation must execute async)
8. Task<TResult> based method which completes async and the returned Task denies child attach (continuation must execute async)

And any other variant that might cause difficulties with the callback being executed.

Reply:
I will not make any changes but will add a comment with a link to your test cases